### ISSUE TITLE
Bundler 1.10.0 breaks `overcommit -r`

### ISSUE THREAD

**Issue by hjblok** [brigade/overcommit#224]
Since 1.10.0 Bundler tracks the Bundler version in lockfile (https://github.com/bundler/bundler/pull/3485). The version is also updated if there were no other changes to the `Gemfile.lock`. For example when you use `bundle exec` (https://github.com/bundler/bundler/issues/3697).

This behavior breaks the `Overcommit::Hook::PreCommit::BundleCheck` functionality when this Bundler version is not (yet) committed. For example when using `overcommit -r`.
    
    result = execute(%w[git diff --quiet --] + [LOCK_FILE])

The above line fails because the `Gemfile.lock` contains uncommitted changes. See [lib/overcommit/hook/pre_commit/bundle_check.rb#L19](https://github.com/brigade/overcommit/blob/master/lib/overcommit/hook/pre_commit/bundle_check.rb#L19).

**Comment by sds** [comment]
Thanks for bringing this issue to our attention, @hjblok.

The discussion in https://github.com/bundler/bundler/issues/3697 suggests that while this behavior is inconvenient it's not going away, so it seems like the best Overcommit can do is [adopt the process endorsed by the Bundler team](https://github.com/bundler/bundler/issues/3697#issuecomment-107226816) and encourage you to update the Bundler version in your `Gemfile.lock`.

In other words, it seems like Overcommit is acting correctly in this case, since it's pointing out that your lockfile is not up to date due to the Bundler version difference.
